.github/workflows/publish.yaml

```diff
@@ -0,0 +1,138 @@
+name: Publish Python 🐍 distribution 📦 to PyPI and TestPyPI
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+
+    steps:
+      # Step 1: Checkout the first repository
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Step 2: Set up Node.js
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20" # Specify the Node.js version to use
+
+      # Step 3: Install openapi-merge-cli and openapi-generator-cli
+      - name: Install openapi-merge-cli
+        run: |
+          npm i openapi-merge-cli
+          npm i @openapitools/openapi-generator-cli
+
+      # Step 4: Set up Python
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+
+      # Step 5: Install Python dependencies
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install build
+
+      # Step 6: Run the Python script to generate json for openApi-merge
+      - name: generate json for openApi-merge
+        run: python3 ./python-client/bin/json_merge_generator.py
+
+      # Step 7: Merge the YAML files
+      - name: Merge the yaml files
+        run: npx openapi-merge-cli --config ./openapi-merge.json
+
+      # Step 8: Generate the SDK
+      - name: Generate the SDK
+        run: npx @openapitools/openapi-generator-cli generate -i ./wordliftApiSpec.yaml -o ./wordlift-client -g python --package-name 'wordlift_client' --library 'asyncio' --additional-properties='generateSourceCodeOnly=true'
+
+      # Step 9: Build the binary
+      - name: Build a binary wheel and a source tarball
+        run: |
+          cd ./wordlift-client
+          python3 -m build
+      - name: Store the distribution packages
+        uses: actions/upload-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist/
+
+  test:
+    name: Test client
```

Comment:
Did you see the tests in the generated GH Action? Is it useful, should we copy them over here:

wordlift-client/.github/workflows/python.yml

Reply:
yes we already have testing in the publish yaml but it make sense to have an action for testing that it's not related to building and publishing and that is triggered also for pushes not on main